### PR TITLE
Dev: help: support multi level subcommands

### PR DESCRIPTION
**Pull request by nicholasyang2022**
#1471 adds 3rd level subcommands to crmsh. However, the `help` module is hardcoded to uses 2 levels of subcommands. This pull request modifies the `help` module to support arbitrary levels of subcommands.

It consists of 2 parts:

1. Replace `_`, the subcommand level separators used in `crm.8.adoc`, with `.`, to avoid ambiguity with `_` used in subcommand names (such as `geo_init`).
2. Use tree structure to store help data in memory, so that it works with arbitrary levels of subcommands.

### CODE REVIEW

File: crmsh/help.py [hunk 1]
```diff
@@ -129,6 +125,48 @@ def __repr__(self):
...
+            self,
+            short_help: str,
+            cmd_args: typing.Sequence[str],
+            alias_for=None, generated=False,
+    ):
+        super().__init__(short_help, '', alias_for, generated)
+        self._cmd_args = cmd_args
+
+    @functools.cached_property
+    def long(self):
```
Comment (liangxin1300) [inline crmsh/help.py:139]: rename as `long_help` might look more descriptive

File: crmsh/help.py [hunk 2]
```diff
@@ -129,6 +125,48 @@ def __repr__(self):
...
+        args = ['crm']
+        args.extend(self._cmd_args)
+        args.append('--help-without-redirect')
+        rc, stdout = ShellUtils.get_stdout(args, shell=False)
```
Comment (liangxin1300) [inline crmsh/help.py:143]: use `get_stdout_stderr` should be better for storing error msg

File: crmsh/help.py [hunk 3]
```diff
@@ -129,6 +125,48 @@ def __repr__(self):
         return str(self)
 
 
+class LazyHelpEntryFromCli(HelpEntry):
```
Comment (liangxin1300) [inline crmsh/help.py:128]: doc string needed

File: crmsh/help.py [hunk 4]
```diff
@@ -129,6 +125,48 @@ def __repr__(self):
...
+
+
+@dataclasses.dataclass
+class SubcommandTreeNode:
+    name: str
+    help: HelpEntry
+    children: dict[str, typing.Self]
+
+    def is_leaf(self) -> bool:
+        return bool(self.children)
```
Comment (liangxin1300) [inline crmsh/help.py:157]: Typically, a leaf node is a node with, or without children?

File: crmsh/help.py [hunk 5]
```diff
@@ -129,6 +125,48 @@ def __repr__(self):
...
+        args.append('--help-without-redirect')
+        rc, stdout = ShellUtils.get_stdout(args, shell=False)
+        return stdout
+
+    def paginate(self):
+        page_string('{}\n\n{}'.format(self.short, self.long))
+
+
+@dataclasses.dataclass
+class SubcommandTreeNode:
```
Comment (liangxin1300) [inline crmsh/help.py:151]: docstring needed for this new class

File: crmsh/help.py [hunk 6]
```diff
@@ -163,37 +204,34 @@ def get_max_width(dict_):
     return max_width
 
 
+def _render_commmad_tree(out: io.StringIO, node: SubcommandTreeNode, indent: int):
```
Comment (liangxin1300) [inline crmsh/help.py:207]: typo here `commmad`

File: crmsh/help.py (hunk 4 above)
Comment (nicholasyang2022) [reply crmsh/help.py]: I removed this method as it is not used. In recursions, it is more straightforward to loop over `children` directly without checking whether it is a leaf node.

**Review by liangxin1300 — APPROVED** [review]
Great work! Thanks!

**Pull request merged**